### PR TITLE
Config polling

### PR DESCRIPTION
**Pull request by j-chmielewski**
Implements periodic polling for configuration updates.

### CODE REVIEW

File: src-tauri/src/periodic/config.rs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    let request = build_request(pool, instance).await?;
+    let url = Url::from_str(&instance.proxy_url)
+        .and_then(|url| url.join(POLLING_ENDPOINT))
+        .map_err(|_| {
+            Error::InternalError(format!(
+                "Can't build polling url: {}/{}",
+                instance.proxy_url, POLLING_ENDPOINT
+            ))
+        })?;
+    let response = reqwest::Client::new().post(url).json(&request).send().await;
```
Comment (t-aleksander) [inline src-tauri/src/periodic/config.rs:75]:
```suggestion
    let response = reqwest::Client::new().post(url).json(&request).timeout(Duration::from_secs(5)).send().await;
```
maybe? Not sure if there is any default timeout:
https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.timeout
https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.timeout

File: src-tauri/src/commands.rs [hunk 2]
```diff
@@ -73,12 +78,13 @@ pub async fn disconnect(
...
         debug!("Connection saved");
         handle.emit_all(
-            "connection-changed",
+            CONNECTION_CHANGED,
             Payload {
                 message: "Created new connection".into(),
             },
         )?;
         stop_log_watcher_task(&handle, &interface_name)?;
+        maybe_update_instance_config(location_id, &handle).await?;
```
Comment (t-aleksander) [inline src-tauri/src/commands.rs:87]:
```suggestion
        if connection_type == ConnectionType::Location {
            maybe_update_instance_config(location_id, &handle).await?;
        }
```

seems like passing tunnel id as location_id will always result in an error

File: src-tauri/src/periodic/config.rs [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    loop {
+        let Ok(instances) = Instance::all(&pool).await else {
+            error!("Failed to retireve instances, retrying in {}s", INTERVAL_SECONDS.as_secs());
+            sleep(INTERVAL_SECONDS).await;
+            continue;
+        };
+        debug!(
+            "Polling configuration updates for {} instances",
+            instances.len(),
+        );
```
Comment (t-aleksander) [inline src-tauri/src/periodic/config.rs:35]:
```suggestion
        if instances.is_empty() {
            debug!(
                "No instances found, skipping configuration polling, sleeping {}s",
                INTERVAL_SECONDS.as_secs()
            );
            sleep(INTERVAL_SECONDS).await;
            continue;
        }
        debug!(
            "Polling configuration updates for {} instances",
            instances.len(),
        );
```

maybe we should check if there are any instances at all? If someone uses the client purely for the tunnel functionality (as a "vanilla" wg client) the info log "Retrieved configuration for {} instances, sleeping {}s" would be still displayed every 30s

**Review by t-aleksander — CHANGES_REQUESTED** [review]

File: src-tauri/src/periodic/config.rs (hunk 3 above)
Comment (j-chmielewski) [reply src-tauri/src/periodic/config.rs:35]: I think it's fine, I'll take it over repeated sleep & continue.

**Pull request merged**